docs/generalized_distribution.md

```diff
@@ -0,0 +1,83 @@
+
+Once the original partition receives this message it writes the `CommandDistribution.ACKNOWLEDGE`
+command to the log. When processing this it knows the distribution to a specific partition is
+completed. Once all partitions acknowledged the distribution the `CommandDistribution.FINISHED`
+event is written. This indicates the finalization of the distribution. At this point all the record
+metadata and raw record value are removed from the state.
+
+### The `CommandDistributionRecord`
+
+The diagram shows the flow of commands and events. What it doesn't highlight is the record that
+belongs to these.
+
+The `CommandDistributionRecord` is relatively simple. It contains 4 properties which are required to
+distribute a command successfully.
+
+|   Property    |                                                                                           Description                                                                                            |
+|---------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Partition id  | The value of the partition id is the id of the partition that has created the record.                                                                                                            |
+| Value type    | The value is part of the record so we know what type of record we are distributing. We need this in combination with the command value to recreate the distributed record.                       |
+| Intent        | The intent is required during redistribution of the command. We must know what intent we should distribute the command with. During normal distribution we can get this from the command itself. |
+| Command value | The command value is the record which is getting distributed. This contains the actual data required by the processor on the other partition.                                                    |
+
+### Redistributing
+
+A command distribution means we have network traffic between partition. Where there's network
+traffic there's a chance of failure. If this happens we need a way to redistribute the command. For
+this we have the `CommandRedistributor`.
+
+As shown in the diagram we store multiple things in the state. First there's the record metadata and
+the raw record value. Next there's a pending command distribution. Both of these are used during
+redistribution of a command.
+
+Every 10 seconds we check if there are any pending command distributions. If there are it means we
+did not receive an acknowledgement for one of our distributions. We will read the record metadata
+and the raw record value from the state and resend the distribution in hopes we will get an
+acknowledgement this time.
+
+**Note!** It is important to realise that this could result in a partition receiving a distribution
```

Comment:
🔧 If you remove the exclamation mark, GH renders this as a note:
```suggestion
**Note** It is important to realise that this could result in a partition receiving a distribution
```

>**Note!** It is important to realise that this could result in a partition receiving a distribution

>**Note** It is important to realise that this could result in a partition receiving a distribution